### PR TITLE
Added nix 1.11.5

### PR DESCRIPTION
**Pull request by zimbatm**
Hi, this is my first cask package. Let me know if I missed anything.

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked that the cask was not already refused in [closed issues].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed

### CODE REVIEW

File: Casks/nix.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  url "https://nixos.org/releases/nix/nix-#{version}/nix-#{version}-x86_64-darwin.tar.bz2"
+  name 'nix'
+  homepage 'https://nixos.org/nix/'
+
+  depends_on macos: '>= :yosemite'
+
+  installer script: "nix-#{version}-x86_64-darwin/install", sudo: false
+
+  uninstall delete: [
```
Comment (vitorgalvao) [inline Casks/nix.rb]: Did this pass `style` as it was? I find that weird.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
This seems to be a CLI-only open-source tool. Please submit it first to [Homebrew](https://github.com/Homebrew/homebrew-core) and only if they don’t accept it, submit here (linking back to the pull request where it was refused).

This policy is documented in [faq/rejected_casks](https://github.com/caskroom/homebrew-cask/blob/master/doc/faq/rejected_casks.md):

> The app is both open-source and CLI-only (i.e. it only uses the `binary` artifact). In that case, and [in the spirit of deduplication](https://github.com/caskroom/homebrew-cask/issues/15603), submit it first to [Homebrew](https://github.com/Homebrew/homebrew). If it is rejected there, you may then try again in Homebrew-Cask (link us to the issue on Homebrew so we can see their reasoning for rejection).

File: Casks/nix.rb (hunk 1 above)
Comment (zimbatm) [inline Casks/nix.rb]: Yes it did

**Comment by zimbatm** [comment]
Thanks a lot for reviewing and fixing my code.

nix is a package manager itself, just like macports which is also in this repo. It's quite unlikely that I will be able to build it from source and propose a recipe to the main homebrew repo as it's build process depends on nix itself. Right now it's bootstrapped from another system that has nix installed already.

**Comment by miccal** [comment]
> nix is a package manager itself, just like macports

Yes but the Cask for [MacPorts](https://github.com/caskroom/homebrew-cask/blob/master/Casks/macports.rb) installs a `.pkg` as its `artifact`, which is totally different to how Nix is installed.

**Comment by zimbatm** [comment]
The Macports CLI could also probably be compiled using a Homebrew formula. After the installation I think both have the same set of constraints that don't make them good candidates for Homebrew.

They both install under a non-standard prefix. Macports uses `/opt/local` and Nix uses `/nix`. I believe that Homebrew wants to keep everything under `/usr/local`.

They both self-upgrade and want to manage software themselves. Having Homebrew hold one part of the software under `/usr/local` would probably cause issues. http://docs.brew.sh/Acceptable-Formulae.html also says "We don’t like tools that upgrade themselves"

In that sense I see Homebrew Cask most like a bootstrap in both cases. Basically if users already have Homebrew installed but need extra dependencies from macports or nix then it makes it easier to extend the package set. They could also run some `curl | sh` script but having this managed is nice to have and also helps by establishing the trust chain.

**Comment by vitorgalvao** [comment]
> They could also run some `curl | sh` script but having this managed is nice to have and also helps **by establishing the trust chain**.

[emphasis added]

**I really hope not**, with the utmost sincerity. For one, we’re not actually verifying the script, just running it, so there’s absolutely no reason users should trust this more simply because it comes from us.

In addition, we’re adamant users do not trust us blindly: see https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment and https://github.com/caskroom/homebrew-cask/blob/master/doc/faq/not_a_discoverability_service.md.

The rest of the argument makes sense, though.

File: Casks/nix.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  url "https://nixos.org/releases/nix/nix-#{version}/nix-#{version}-x86_64-darwin.tar.bz2"
+  name 'nix'
+  homepage 'https://nixos.org/nix/'
+
+  depends_on macos: '>= :yosemite'
+
+  installer script: "nix-#{version}-x86_64-darwin/install",
+            sudo:   false
+
+  uninstall delete: [
```
Comment (vitorgalvao) [inline Casks/nix.rb]: Is this really enough to completely remove it?

Comment (vitorgalvao) [inline Casks/nix.rb]: We don’t have upgrade functionality.

**Review by vitorgalvao — APPROVED** [review]

File: Casks/nix.rb (hunk 2 above)
Comment (zimbatm) [inline Casks/nix.rb]: Okay.

I added the cleanup of it in 75658532d570df6375aa6a464c0b2629d7474702

**Comment by zimbatm** [comment]
Thanks @vitorgalvao. It's the first time I'm cooperating on a PR with the reviewer pushing code to it and it's quite nice actually, it feels much more collaborative like that.

Do you want me to squash and rebase the PR?

**Comment by miccal** [comment]
There are a few `style` errors:
```
>>> brew cask style Casks/nix.rb
== Casks/nix.rb ==
C: 14:  3: uninstall stanza out of order
C: 16:  3: uninstall_postflight stanza out of order
C: 20:  7: Use next to skip iteration.
C: 26: 33: Align the elements of an array literal if they span more than one line.
```

**Comment by zimbatm** [comment]
All fixed now

**Review by adidalal — APPROVED** [review]

**Pull request merged**